smithyctl/types/v1/component.go

```diff
@@ -0,0 +1,44 @@
+package v1
+
+import (
+	"github.com/google/uuid"
+)
+
+type (
+	// ComponentType represents all the types of components that Smithy supports.
+	// ENUM(unknown, target, scanner, enricher, filter, reporter)
+	ComponentType string
+
+	// Component is a binary or a script that can be used in the context of a
+	// workflow to generate some useful result.
+	Component struct {
+		// Description describes what the component will do.
+		Description string
+		// ID is a unique identifier for a component.
+		ID uuid.UUID
```

Comment:
I don't think we will need t a UUID for the component because it's not stored in a database and there is no central authority to check the uniqueness of this ID